### PR TITLE
Add Resolves/Closes Keywords and updated PR Template

### LINKED ISSUE #343: Add Resolves/Closes Keywords to PR Template
#### Reason/Context

Currently, when a PR is created for an issue, the PR template includes `GitHub Issue:`, which only mentions the original issue in the PR.

It would be really helpful if the issues are connected to the PR, which will improve clarity and connect issues to PR. This way, when issues are mentioned in their PR, they will be connected, and the issues will be automatically closed upon merging the PR. Moreover, issues tab will display if a PR is created for the particular issue. 

#### Description

Introduce the use of `Resolves` or `Closes` keyword in the PR template. This improvement aims to establish a connection between mentioned issues and pull requests.

For reference:
https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue

**Do you think this work might require an [Architectural Decision Record (ADR)](https://github.com/json-schema-org/community/blob/main/CONTRIBUTING.md#key-architectural-decisions)? (significant or noteworthy)**

No

### PR DESCRIPTION
**Pull request by aialok**
**What kind of change does this PR introduce?**

- Added PR dependencies in workflow, Improve PR-Template.

**Issue Number: #343 
Resolves: #343 

- Closes #343 
- Related to # NA
- Others? NA

**Snapshots:**
- NA

**If relevant, did you update the documentation?**
- NA

**Summary**
- Added : 
```md
-  Closes #___ <!-- Replace ___ with the issue number this PR resolves -->
-  Related to #___ <!-- Use when the PR doesn't completely resolve an issue -->
-  Others? <!-- Add any additional notes or references here -->

```
- Updated PR-Template for better review.
- Added PR Dependencies 

**Does this PR introduce a breaking change?**

- No

**Do you think resolving this issue might require an [Architectural Decision Record (ADR)](https://github.com/json-schema-org/community/blob/main/CONTRIBUTING.md#key-architectural-decisions)? (significant or noteworthy)**

Yes/No
[JUSTIFICATION]

### CODE REVIEW

File: .github/workflows/pr-dependencies.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+name: Check PR Dependencies
```
Comment (gregsdennis) [inline .github/workflows/pr-dependencies.yml:1]: This is already an org-wide action (as I had stated in the issue).  It's not needed explicitly here.  Please remove.

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -1,13 +1,40 @@
-
 <!-- In order to keep off topic discussion to a minimum, it helps if the "work to be done" is already agreed on. -->
```
Comment (gregsdennis) [inline .github/pull_request_template.md:1]: https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/creating-a-pull-request-template-for-your-repository

Path should be `.github/PULL_REQUEST_TEMPLATE/[filename].md`  (pick a file name).

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -1,13 +1,40 @@
...
+-->
+
+**What kind of change does this PR introduce?**
+
+<!-- E.g. a bugfix, feature, refactoring, etc… -->
+
+**Issue Number:**
+
+- Closes # <!-- Replace ___ with the issue number this PR resolves -->
+- Related to # <!-- Use when the PR doesn't completely resolve an issue -->
```
Comment (gregsdennis) [inline .github/pull_request_template.md:14]: Should update the comments. There is no `___`.

File: .github/pull_request_template.md [hunk 4]
```diff
@@ -1,13 +1,40 @@
...
+- Others? <!-- Add any additional notes or references here -->
+
+**Snapshots:**
```
Comment (gregsdennis) [inline .github/pull_request_template.md:17]: I think "screenshots" is more common.

File: .github/pull_request_template.md [hunk 5]
```diff
@@ -1,13 +1,40 @@
...
+<!--Add snapshots or videos wherever possible.-->
 
-**Summary**:
+**If relevant, did you update the documentation?**
+
+<!--Add link to it-->
+
+**Summary**
+
+<!-- Explain the **motivation** for making this change. What existing problem does the pull request solve? -->
```
Comment (gregsdennis) [inline .github/pull_request_template.md:27]: Why is "motivation" bolded?

File: .github/pull_request_template.md [hunk 6]
```diff
@@ -1,13 +1,40 @@
...
+<!-- Try to link to an open issue for more information. -->
+
+**Does this PR introduce a breaking change?**
```
Comment (gregsdennis) [inline .github/pull_request_template.md:30]: Not really sure how there could be a breaking change on a website.  Broken link, maybe?  It doesn't really have programmatic users (where a breaking change might need to be considered).

I'd say we remove this, but can keep it if others think we should.

File: .github/pull_request_template.md [hunk 7]
```diff
@@ -1,13 +1,40 @@
...
+
+<!-- If this PR introduces a breaking change, please describe the impact and a migration path for existing applications. -->
 
 **Do you think resolving this issue might require an [Architectural Decision Record (ADR)](https://github.com/json-schema-org/community/blob/main/CONTRIBUTING.md#key-architectural-decisions)? (significant or noteworthy)**
+
 <!-- If the issue has the `adr-required`, this PR must include an ADR. -->
```
Comment (gregsdennis) [inline .github/pull_request_template.md:36]: I'm not sure if we need ADRs in the website.  We introduced them to track spec decisions.

@benjagm your thoughts?

**Review by gregsdennis — CHANGES_REQUESTED** [review]

**Comment by aialok** [comment]
Making the desire changes.
Thank you @gregsdennis for your review : )

File: .github/workflows/pr-dependencies.yml (hunk 1 above)
Comment (aialok) [reply .github/workflows/pr-dependencies.yml:1]: Okay I will be removing it.

File: .github/pull_request_template.md (hunk 6 above)
Comment (aialok) [reply .github/pull_request_template.md:30]: If in future If we have unit testing then might be this is in use.

File: .github/pull_request_template.md (hunk 5 above)
Comment (aialok) [reply .github/pull_request_template.md:27]: Sorry, I will fix this. Basically.

File: .github/pull_request_template.md (hunk 2 above)
Comment (aialok) [reply .github/pull_request_template.md]: Done.

File: .github/pull_request_template.md (hunk 7 above)
Comment (benjagm) [reply .github/pull_request_template.md]: Let's remove the ADR section from the template.

**Review by gregsdennis — APPROVED** [review]

**Review by benjagm — APPROVED** [review]
LGTM!

**Pull request merged**